### ISSUE TITLE
3.Estimarea costului total pe baza costurilor FN/FP. Ștefania(Gr.B)

### ISSUE THREAD

**Issue by licastefania200001-alt** [georgerusu017/churn_rate_predicition#51]

**Comment by licastefania200001-alt** [comment]
import pandas as pd
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, average_precision_score

df = pd.read_csv("tranzactii.csv", sep="\t")

df = df.dropna(subset=['Class'])

df['Class'] = df['Class'].astype(int)

y_true = df['Class']

y_score = df[['f1','f2','f3','f4','f5']].sum(axis=1)

threshold = 0.5
y_pred = (y_score >= threshold).astype(int)

cm=confusion_matrix(y_true, y_pred)
print("matrice de confuzie")
print(cm)
print("Accuracy :", accuracy_score(y_true, y_pred))
print("Precision:", precision_score(y_true, y_pred, zero_division=0))
print("Recall:", recall_score(y_true, y_pred, zero_division=0))
print("PR-AUC:", average_precision_score(y_true, y_score))


cost_FN=1000
cost_FP=100
tn, fp, fn, tp= cm.ravel()
cost_total=fn * cost_FN + fp * cost_FP
print(f"Cost total: {cost_total}")
